src/gui.cpp

```diff
@@ -60,15 +62,41 @@ Gui::Gui(QMainWindow *win, Ui_GUI *ui_win, Gallery *galleryIn, Camera *camera, s
 
     });
 
-    QObject::connect(ui->lineEdit, &QLineEdit::textChanged, ui->edgeEnhancementSlider, [&](const QString &text) {
+    QObject::connect(ui->edgeEnhancementValueInput, &QLineEdit::textChanged, ui->edgeEnhancementSlider, [&](const QString &text) {
         bool ok;
         int value = text.toInt(&ok);
         if (ok) {
             ui->edgeEnhancementSlider->setValue(value);
         }
     });
     
+    // //-----------block -2 contrast ---------------------
+    QObject::connect(ui->contrastEnhancementSlider, &QSlider::valueChanged, ui->contrastEnhancementValueInput, [&](int sliderValue1) {
+        ui->contrastEnhancementValueInput->setText(QString::number(sliderValue1));
+        bool enabled = blocks[blocks.size()-2]->getEnabled();
```

Comment:
Maybe change the -2 to the normal index, eg block 3 in this case and move it above edge enhancement to make it clear and easy to follow. Edge enhancement is always the last block in the chain.